### PR TITLE
Permanent failure message for letters

### PR DESCRIPTION
**Pull request by servingUpAces**
When a letter has passed our validation but the postal provider is unable to print the letter we need to mark the letter as failed.

If we mark the letter as a technical-failure, we say that we will fix the issue, which is wrong because we cannot fix the issue.

If we mark the letter as validation-failed, the letter is in the wrong bucket, so the letter is not viewable/downloadable by the client.

This PR updates the message for a letter marked as permanent-failure to better reflect what has actually happened.

### CODE REVIEW

File: app/templates/views/notifications/notification.html [hunk 1]
```diff
@@ -41,14 +41,18 @@
...
         <p class="notification-status-cancelled">
           Cancelled {{ updated_at|format_datetime_short }}
         </p>
       {% elif notification_status == 'validation-failed' %}
         <p class="notification-status-cancelled">
           {{ message.summary | safe }}
         </p>
+      {% elif notification_status == 'permanent-failure' %}
+         <p class="notification-status-cancelled">
+          Permanent failure – The postal provider is unable to print the letter. Your letter has not been sent.
```
Comment (karlchillmaid) [inline app/templates/views/notifications/notification.html:54]:
```suggestion
          Permanent failure – The provider cannot print the letter. Your letter will not be dispatched.
```

File: tests/app/main/views/test_notifications.py [hunk 2]
```diff
@@ -442,7 +442,7 @@ def test_notification_page_shows_validation_failed_precompiled_letter(
 @pytest.mark.parametrize('notification_status, expected_message', (
     (
         'permanent-failure',
-        'Cancelled 1 January at 1:02am',
+        'Permanent failure – The postal provider is unable to print the letter. Your letter has not been sent.',
```
Comment (karlchillmaid) [inline tests/app/main/views/test_notifications.py:445]:
```suggestion
        'Permanent failure – The provider cannot print the letter. Your letter will not be dispatched.',
```

File: app/templates/views/notifications/notification.html (hunk 1 above)
Comment (karlchillmaid) [reply app/templates/views/notifications/notification.html:54]: The way we currently describe it, we use:
* ‘sent’ to mean the user sends the letter using Notify
* ‘dispatched’ to mean the provider posts the letter

I just want to check I’ve got the usage right here.

Comment (servingUpAces) [reply app/templates/views/notifications/notification.html]: right - I've updated this message and the status doc page and tech docs

File: app/templates/views/message-status.html [hunk 3]
```diff
@@ -80,6 +80,7 @@ <h2 id="letter-statuses" class="heading-medium">Letters</h2>
         ('Printed', 'The provider has printed the letter. Letters are printed at 5:30pm and dispatched the next working day.'), 
         ('Cancelled', 'Sending cancelled. Your letter will not be printed or dispatched.'),
         ('Technical failure', 'Notify had an unexpected error while sending the letter to our printing provider.'),
+        ('Permenent failre', 'The provider cannot print the letter. Your letter will not be dispatched.')
```
Comment (karlchillmaid) [inline app/templates/views/message-status.html:83]:
```suggestion
        ('Permanent failure', 'The provider cannot print the letter. Your letter will not be dispatched.')
```

File: app/templates/views/message-status.html [hunk 4]
```diff
@@ -80,6 +80,7 @@ <h2 id="letter-statuses" class="heading-medium">Letters</h2>
         ('Printed', 'The provider has printed the letter. Letters are printed at 5:30pm and dispatched the next working day.'), 
         ('Cancelled', 'Sending cancelled. Your letter will not be printed or dispatched.'),
         ('Technical failure', 'Notify had an unexpected error while sending the letter to our printing provider.'),
+        ('Permenent failure', 'The provider cannot print the letter. Your letter will not be dispatched.')
```
Comment (karlchillmaid) [inline app/templates/views/message-status.html:83]:
```suggestion
        ('Permanent failure', 'The provider cannot print the letter. Your letter will not be dispatched.')
```

**Review by leohemsted — APPROVED** [review]

**Comment by klssmith** [comment]
Might need to update this page too (the bottom letter has permanent-failure status)

Screenshot 2021-06-11 at 15 04 18

**Review by klssmith — APPROVED** [review]

**Pull request merged**